### PR TITLE
Upgrade the Cython version whenever we can

### PR DESCRIPTION
**Pull request by lidizheng**
Fixes https://github.com/grpc/grpc/issues/21450, which is a breakage introduced by #21232.

This is a straight forward error that the Cython version is too old to understand new Python syntax. Once we upgrade Cython package, the problem should go away.

### CODE REVIEW

**Comment by lidizheng** [comment]
[prod:grpc/core/master/linux/grpc_build_artifacts_extra](http://sponge/f38bd2a1-016d-4a0d-88ab-5fa43f90a43b)

**Comment by lidizheng** [comment]
[prod:grpc/core/master/linux/grpc_build_artifacts_extra](http://sponge/d08c8826-4119-4232-8dbb-b282451e1b44)

File: tools/run_tests/artifacts/build_artifact_python.sh [hunk 1]
```diff
@@ -22,6 +22,9 @@ export PYTHON=${PYTHON:-python}
 export PIP=${PIP:-pip}
```
Comment (gnossen) [inline tools/run_tests/artifacts/build_artifact_python.sh]: Is there an equivalent Windows build that should be updated as well?

Comment (lidizheng) [reply tools/run_tests/artifacts/build_artifact_python.sh]: Yes, I have done it in #21340. I assumed Linux would be in good shape, but no.

**Comment by lidizheng** [comment]
Known failures: https://github.com/grpc/grpc/issues/19627

**Pull request merged**